serializer_impl.hh

```diff
@@ -840,7 +884,7 @@ public:
         using reference = value_type&;
         using difference_type = ssize_t;
 
-        iterator() noexcept : _in(simple_input_stream()) {}
+        iterator() noexcept : _data{} {}
```

Comment:
can now be `iterator() = default` since both iterator data classes seem to be default initializeable